iosApp/iosApp/ContentView.swift

```diff
@@ -88,17 +90,70 @@ struct ContentView: View {
         }
     }
 
+    struct LocationAuthButtonStyle: ButtonStyle {
+        func makeBody(configuration: Configuration) -> some View {
+            configuration.label
+                .padding()
+                .background(Color.key)
+                .foregroundStyle(.white)
+                .clipShape(Capsule())
+        }
+    }
+
+    struct LocationAuthLabelStyle: LabelStyle {
+        func makeBody(configuration: Configuration) -> some View {
+            HStack(alignment: .center, spacing: 8) {
+                configuration.title
+                configuration.icon
+            }
+            .fontWeight(.bold)
+        }
+    }
+
     @ViewBuilder
-    var locationAuthHeader: some View {
+    var locationAuthButton: some View {
         switch locationDataManager.authorizationStatus {
-        case .notDetermined:
-            Button("Allow Location", action: {
-                locationDataManager.locationFetcher.requestWhenInUseAuthorization()
+        case .notDetermined, .denied, .restricted:
+            Button(action: {
+                showingLocationPermissionAlert = true
+            }, label: {
+                Label(title: {
+                    Text("Location Services is off")
+                }, icon: {
+                    Image(.faChevronRight)
+                        .resizable()
+                        .scaleEffect(0.6)
+                        .frame(width: 20, height: 20, alignment: .center)
+                        .backgroundStyle(.white)
+                        .background(in: .circle)
+                        .foregroundColor(Color.key)
+                })
+                .labelStyle(LocationAuthLabelStyle())
             })
+            .buttonStyle(LocationAuthButtonStyle())
+            .accessibilityIdentifier("locationServicesButton")
+            .alert(
+                "Maps work best with Location Services turned on",
+                isPresented: $showingLocationPermissionAlert,
+                actions: {
+                    Button("Turn On in Settings") {
+                        showingLocationPermissionAlert = false
+                        if let url = URL(string: UIApplication.openSettingsURLString) {
+                            UIApplication.shared.open(url)
+                        }
+                    }
+                    Button("Keep Location Services off") {
+                        showingLocationPermissionAlert = false
+                    }
+                },
+                message: {
+                    Text(
+                        "You'll get turn-by-turn directions, estimated travel times, and improved search results when you turn on Location Services for Maps"
```

Comment:
issue: this text should be "You’ll see nearby transit options and get better search results when you turn on Location Services for MBTA Go.", we don't offer turn-by-turn directions (at least not yet!)

Reply:
🤦 Sorry, I was looking at the image rather than the text.